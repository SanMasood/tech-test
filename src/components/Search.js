import React, {useState} from 'react';

import '../styles/Search.css'


const Search = () => {
return (
    
    <div className="Search">
        <img src="https://lofrev.net/wp-content/photos/2014/09/Nasa-logo-3D.gif" className="nasa-logo" alt="NASA-logo"/>
        <input className="search-text" type="text" />
    </div>
)
}

export default Search;