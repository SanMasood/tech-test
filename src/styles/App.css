.App {
    display: flex;
    flex-direction: column;
    align-content: center;
    margin: auto;
   
    height: 100%;
    background: linear-gradient(rgb(231, 231, 253), rgb(243, 236, 236));
}

.nasa-logo {
    margin: auto;
    padding: 5px;
    width: 200px;
    max-height: 250px;
}

.results-text {
    text-align: center;
}