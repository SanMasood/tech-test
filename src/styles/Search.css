.Search{
    display: flex;
    flex-direction: column;
    margin-top: 1rem;
}

.nasa-logo{
    height: 20%;
    width: 20%;
    margin: 0 auto;
    padding-bottom: 1rem;
}

.search-text{
    width: 40%;
    margin: 0 auto;
}