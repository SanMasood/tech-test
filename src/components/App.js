import React, { useState } from 'react';

import Search from './Search';
import SearchResults from './SearchResults';
import logoImage from '../assets/nasa-icon-png-7.png'

import '../styles/App.css';
import '../styles/Search.css'
import '../styles/SearchResults.css'

function App() {
  const [searchResults, setSearchResults] = useState([]);
  //searchResults.map((e)=> console.log(e));
  


  return (
    <div className="App">
       <img src={logoImage} data-testid="logo-id" className="nasa-logo" alt="NASAlogo"/>
       
        <Search setSearchResults = { setSearchResults }/>

        <div className="search-results">
          {
                   
           (searchResults && searchResults.length > 0)?
           searchResults.map(image => <SearchResults image = {image} /> ):
           <div><h2 className="results-text">No results yet.</h2></div>              
           
          }
        </div>
       
    </div>
  );
}



export default App;
