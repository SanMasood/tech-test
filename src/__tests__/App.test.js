import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../components/App';
import Search from '../components/Search';

import logoImage from '../assets/Nasa-logo-3D.gif'


describe ("App", () => {
  it("renders correctly", () => {
    const {asFragment} = render (
      <App 
      />
    )
    screen.getByText("Search:");
    expect(asFragment()).toMatchSnapshot();
    })


    it("Has Search label", () => {
        render(<Search />);
        expect(screen.getByText(/Search/)).toBeInTheDocument();
    });


    it('displays the correct logo', () => {
        const { getByAltText }= render(<img src={logoImage} alt="NASAlogo"/>)
          
        expect(screen.getByAltText("NASAlogo")).toBeInTheDocument();

    });

    it ("renders the button", () => {
    const { getByRole } = render(<button />)

    expect (screen.getByRole('button')).toBeInTheDocument();

    });

    it ('renders textbox to type in', () => {
        const {getByRole} = render(<input  />)       
        //screen.debug();   
        expect(screen.getByRole('textbox')).toBeInTheDocument();
    })
})
