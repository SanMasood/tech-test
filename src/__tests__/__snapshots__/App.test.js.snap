// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="Search"
    >
      <img
        alt="NASA-logo"
        class="nasa-logo"
        src="https://lofrev.net/wp-content/photos/2014/09/Nasa-logo-3D.gif"
      />
      <input
        class="search-text"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
